resources/security_group.rb

```diff
@@ -0,0 +1,255 @@
+#
+# Cookbook:: aws
+# Resource:: security_group
+#
+# Technically managing the security group and the ingress/egress
+# rules are unique API calls.  It's much easier to handle them in
+# a single provider, partially to limit the number of API calls.
+# Mostly because describing a security group returns the entire object
+# which we can use to compare against our object
+
+resource_name :security_group
+provides :aws_security_group
+
+# => Define the Resource Properties
+property :security_group_name, String, name_property: true
+property :description, String, default: 'created by chef'
```

Comment:
should we default to have something that includes the security group name in it?

Reply:
@majormoses - a better option might be to make this a required field with no default.  what do you think?  it's a required field in aws.  I was thinking of adding some code to add the name into the description and it seemed like added code and complexity when we could just ask users to set a description.

Reply:
If aws requires it then I think we should as well. From what I recall these are immutable so its important to get it right the first time.